test/e2e/accounts/create-snap-account.spec.ts

```diff
@@ -102,170 +114,187 @@ describe('Create Snap Account', function (this: Suite) {
         });
 
+        // switch to extension full screen view
         await driver.switchToWindowWithTitle(
           WINDOW_TITLES.ExtensionInFullScreenView,
         );
 
+        // account should be created with the custom name
         await driver.findElement({
           css: '[data-testid="account-menu-icon"]',
-          text: 'Snap Account 1',
+          text: newAccountLabel,
         });
       },
     );
   });
 
-  it('create Snap account confirmation cancelation results in error in Snap', async function () {
+  it('create Snap account confirmation cancellation results in error in Snap', async function () {
     await withFixtures(
       {
         fixtures: new FixtureBuilder().build(),
         ganacheOptions: defaultGanacheOptions,
         title: this.test?.fullTitle(),
       },
       async ({ driver }: { driver: Driver }) => {
-        await unlockWallet(driver);
-        // navigate to test Snaps page and connect
-        await driver.openNewPage(TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL);
-        await driver.clickElement('#connectButton');
-
-        // switch to metamask extension and click connect to start installing the snap
-        await switchToNotificationWindow(driver);
-        await driver.clickElement({
-          text: 'Connect',
-          tag: 'button',
-        });
-
-        // scroll to the bottom of the page
-        await driver.waitForSelector({ text: 'Confirm' });
-        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
-
-        // click the install button to install the snap
-        await driver.waitForSelector({ text: 'Confirm' });
-        await driver.clickElement({
-          text: 'Confirm',
-          tag: 'button',
-        });
-        await driver.waitForSelector({ text: 'OK' });
-        await driver.clickElement({
-          text: 'OK',
-          tag: 'button',
-        });
+        // start the create account flow and switch to dialog window
+        await startCreateSnapAccountFlow(driver);
 
-        // Wait until popup is closed before proceeding
-        await driver.waitUntilXWindowHandles(2);
+        // cancel account creation
+        await driver.clickElement('[data-testid="confirmation-cancel-button"]');
 
-        // move back to the Snap window to test the create account flow
-        await driver.switchToWindowWithTitle(
+        // switch back to the test dapp/Snap window
+        await driver.waitAndSwitchToWindowWithTitle(
+          2,
           WINDOW_TITLES.SnapSimpleKeyringDapp,
         );
 
-        // check the dapp connection status
-        await driver.waitForSelector({
-          css: '#snapConnected',
-          text: 'Connected',
-        });
-
-        // create new account on dapp
-        await driver.clickElement({
-          text: 'Create account',
-          tag: 'div',
+        // account should not be created in Snap
+        await driver.findElement({
+          tag: 'p',
+          text: 'Error request',
         });
```

Comment:
We should probably check that the account has not been created he too, WDYT?

```suggestion
        });
        
        // switch to extension full screen view
        await driver.switchToWindowWithTitle(
          WINDOW_TITLES.ExtensionInFullScreenView,
        );

        // account should not be created
        await driver.assertElementNotPresent({
          css: '[data-testid="account-menu-icon"]',
          text: 'Snap Account 1',
        });
```

Reply:
added in latest commit [2a64b75](https://github.com/MetaMask/metamask-extension/pull/25191/commits/2a64b757ddc7f8cbaa7bf6783173e764e9901786)